### PR TITLE
PR - Report generate 

### PR DESCRIPTION
**Pull request by Leon-JS02**
Upgraded the report generation Lambda.
- Uses a different (simpler) library to build HTML, and generate the HTML to pdf.
- Uses BytesIO to build and send the PDF in memory.
- Added an external styling file for the PDF/HTML report.
- Outsourced the logic to outside the `lambda_handler.py` script. (Simpler file.)

### CODE REVIEW

File: report/generate_report.py [hunk 1]
```diff
@@ -1,130 +1,188 @@
...
-        spaceAfter=12,
-        alignment=0,
-        textColor=colors.black,
-    )
-    return Paragraph(text, heading_style)
+"""File to handle generating of the PDF report bytes object.
+Defines a number of report sections as dynamically generated HTML."""
 
+from datetime import datetime, timedelta
+from io import BytesIO
```
Comment (paulauus) [inline report/generate_report.py:5]: Already looks much simpler than before.

File: report/generate_report.py [hunk 2]
```diff
@@ -1,130 +1,188 @@
...
+    table_html += """
+        </tbody>
+    </table>
+    """
+
+    return table_html
+
+
+def create_style_tag() -> str:
+    """Returns a style tag with CSS for a cute design!"""
```
Comment (paulauus) [inline report/generate_report.py:155]: Cute is good.

File: report/report_style [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<style>
+        body {
+            background-color: #2a2a2a;
+            color: #f5f5f5;
+            font-family: Arial, sans-serif;
```
Comment (paulauus) [inline report/report_style:5]: No need to change the whole thing, but just a reminder that if you want the colours to match the dashboard at any point, you can look up the colour codes in the dashboard folder.

**Review by paulauus — APPROVED** [review]

File: report/report_style
Comment (mehrdadhalali) [inline report/report_style:1]: Nice CSS!

**Review by mehrdadhalali — APPROVED** [review]

**Pull request merged**